DigitalOffice/Tests/ProjectServiceTests/CommandsTests/CreateNewProjectCommandTests.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Net;
+using FluentValidation;
+using Moq;
+using FluentValidation.Results;
+using NUnit.Framework;
+using ProjectService.Commands;
+using ProjectService.Commands.Interfaces;
+using ProjectService.Controllers.RestRequest;
+using ProjectService.Repositories.Interfaces;
+
+namespace ProjectServiceUnitTests.CommandsTests
+{
+    class CreateNewProjectCommandTests
+    {
+        private Mock<ValidationResult> validatorResultMock;
+        private ICreateNewProjectCommand command;
+
+        [SetUp]
+        public void SetUp()
+        {
+            var validatorMock = new Mock<IValidator<NewProjectRequest>>();
+            var repositoryMock = new Mock<IProjectRepository>();
+            validatorResultMock = new Mock<ValidationResult>();
+
+            validatorMock
+                .Setup(x => x.Validate(It.IsAny<IValidationContext>()))
+                .Returns(validatorResultMock.Object);
+
+            command = new CreateNewProjectCommand(repositoryMock.Object, validatorMock.Object);
+        }
+
+        [Test]
+        public void FailCreateNewProjectIncorrectProjectDataTests()
+        {
+            validatorResultMock
+                .Setup(x => x.IsValid)
+                .Returns(false);
+
+            var newProject = new NewProjectRequest
+            {
+                Name = "DigitalOffice",
+                DepartmentId = Guid.NewGuid(),
+                Description = "New project for Lanit-Tercom",
+                IsActive = true,
+                WorkersUsersIds = null,
+                ManagersUsersIds = null,
+                FilesIds = null
+            };
+
+            Assert.Throws<ValidationException>(() => command.Execute(newProject), "Project field validation error");
+        }
+
+        [Test]
+        public void SuccessfulCreateNewProjectTest()
+        {
+            validatorResultMock
+                .Setup(x => x.IsValid)
+                .Returns(true);
+
+            var newProject = new NewProjectRequest
+            {
+                Name = "DigitalOffice",
+                DepartmentId = Guid.NewGuid(),
+                Description = "New project for Lanit-Tercom",
+                IsActive = true,
+                WorkersUsersIds = null,
+                ManagersUsersIds = null,
+                FilesIds = null
+            };
+            
+            Assert.AreEqual(command.Execute(newProject).StatusCode, HttpStatusCode.OK);
```

Comment:
Swap the arguments. In your code expected object is command.Execute(newProject).StatusCode, but it must be HttpStatusCode.OK. What do you think about using FluentAssertions to avoid such mistakes?

Reply:
Fixed